sycl/include/CL/sycl/handler.hpp

```diff
@@ -1257,6 +1281,55 @@ class __SYCL_EXPORT handler {
     } // end while (NWorkItems > 1)
   }
 
+  // This version of parallel_for may handle one or more reductions packed in
+  // \p Rest argument. Note thought that the last element in \p Rest pack is
+  // the kernel function.
+  // TODO: this variant is currently enabled for 2+ reductions only as the
+  // versions handling 1 reduction variable are more efficient right now.
```

Comment:
Out of curiosity, do you have any idea why this is the case?

Reply:
The new implementation on variadic templates does not YET have additional specializations to produce better code for OP/TYPE combinations with 'fast reduce' and/or 'fast atomic fetch'. That is why it is expected to be a bit slower than the old/existing implementation of parallel_for with 1 reduction variable.

Reply:
Ah, I missed that.  Makes sense -- thanks.